pyof/v0x04/common/action.py

```diff
@@ -349,39 +350,42 @@ def __init__(self, action_type=None, ethertype=None):
         self.ethertype = ethertype
 
 
-class ActionSetField(GenericStruct):
+class ActionSetField(ActionHeader):
     """Action structure for OFPAT_SET_FIELD."""
 
-    #: OFPAT_SET_FIELD.
-    action_type = UBInt16(ActionType.OFPAT_SET_FIELD, enum_ref=ActionType)
-    #: Length is padded to 64 bits.
-    length = UBInt16()
-    #: OXM TLV - Make compiler happy
-    field1 = UBInt8()
-    field2 = UBInt8()
-    field3 = UBInt8()
-    field4 = UBInt8()
-
-    def __init__(self, length=None, field1=None, field2=None, field3=None,
-                 field4=None):
+    field = OxmTLV()
+    pad = BinaryData()
+
+    def __init__(self, field=None):
         """Create a ActionSetField with the optional parameters below.
 
         Args:
             length (int): length padded to 64 bits, followed by exactly
                           oxm_len bytes containing a single OXM TLV, then
                           exactly ((oxm_len + 4) + 7)/8*8 - (oxm_len + 4)
                           (between 0 and 7) bytes of all-zero bytes
-            field1 (int): OXM field.
-            field2 (int): OXM field.
-            field3 (int): OXM field.
-            field4 (int): OXM field.
+            field (:class:`OxmTLV`): OXM field and value.
         """
-        super().__init__()
-        self.length = length
-        self.field1 = field1
-        self.field2 = field2
-        self.field3 = field3
-        self.field4 = field4
+        super().__init__(action_type=ActionType.OFPAT_SET_FIELD)
+        self.field = OxmTLV() if field is None else field
+        self.pad = b''
```

Comment:
There's no `pad` attribute in the spec, so we suggest you to follow what we had already done in https://github.com/kytos/python-openflow/blob/698c2825fa89e18e3fa16a9ea05eb048ad56a8a0/pyof/v0x04/common/flow_match.py#L312